site/docs/v1/tech/developer-guide/api-gateways/aws-api-gateway.adoc

```diff
@@ -0,0 +1,217 @@
+
+image::api-gateways/aws-api-gateway/name-lambda.png[Name the Lambda,width=1200]
+
+Under the [breadcrumb]#Code# tab for the lambda, click on the "index.mjs" file and replace the contents with the following code to return the claims found in the JWT:
+
+[source,javascript,title="Lambda function to reflect user claims"]
+----
+export const handler = async(event) => {
+    const claims = event.requestContext.authorizer.claims;
+    const response = {
+        statusCode: 200,
+        body: JSON.stringify(claims),
+    };
+    return response;
+};
+----
+
+Then click the `Deploy` button to update the lambda.
+
+Click the expander next to [breadcrumb]#Function overview# near the top of the page. Then click on `Add Trigger`.
+
+Under [field]#Trigger Configuration#, choose `API Gateway` as the source. Select `Use an existing API` and then select the API gateway set up earlier. Select the `$default` deployment stage.
+
+Under [field]#Security#, choose `Create a JWT authorizer`.
+
+Set [field]#Identity source# to `$request.header.Authorization`.
+
+For [field]#Issuer#, set the URL to be the same as you set in FusionAuth for the JWT issuer in the Tenant configuration. Make sure that it is exactly the same, including trailing slashes, without any whitespace.
+
+Under [field]#Audiences#, paste in the `Client ID` from the FusionAuth application created earlier.
+
+Then click the primary `Add` button at the bottom right of the screen.
+
+After completing this, a new route will be added to your gateway API, with the same name as the lambda, `privateFunction` here. Concat this name with the "Invoke URL" recorded earlier when setting up the API gateway to get the full URL. It should look similar in form to this: `\https://w3miabt10d.execute-api.us-east-2.amazonaws.com/privateFunction`.
+
+If you call this route through a browser, Postman, or cURL, you should receive a 401 and an `Unauthorized` message. We'll create a token next with FusionAuth to show how the route can be successfully called.
+
+== Testing with a token from FusionAuth
+
+To access the secured lambda function, we'll need to get a valid token from FusionAuth. Normally, your frontend app would redirect a user to FusionAuth to login, obtain a code, and then exchange that code for a token, using a framework of your choice. For the purposes of this guide, we'll call the FusionAuth API on behalf of a user to create a token directly.
+
+To enable this, there are a few settings that need to be changed on FusionAuth. In the side panel in FusionAuth, select [breadcrumb]#Settings > API Keys#. Click the green `+` button at the top left to create a new key.
+
+Give the key a meaningful description, like `API Gateway Key`. Select the Tenant that you created the FusionAuth application under (normally `Default`).
+
+Then, under [breadcrumb]#Endpoints#, allow `POST` on the `/api/login` route.
```

Comment:
```suggestion
Then, under [breadcrumb]#Endpoints#, allow `POST` on the `/api/login` route. This key will only be allowed to all this endpoint with the `POST` method.
```

Reply:
*call* this endpoint I guess?

Reply:
Correct! `This key will only be allowed to call`